x/slashing/client/cli/query.go

```diff
@@ -51,16 +52,16 @@ $ <appd> query slashing signing-info cosmosvalconspub1zcjduepqfhvwcmt7p06fvdgexx
 				return err
 			}
 
-			queryClient := types.NewQueryClient(clientCtx)
-
-			pk, err := sdk.GetPubKeyFromBech32(sdk.Bech32PubKeyTypeConsPub, args[0])
-			if err != nil {
+			var pk cryptotypes.PubKey
+			am := codec.NewJSONAnyMarshaler(clientCtx.JSONMarshaler, clientCtx.InterfaceRegistry)
+			if err := codec.UnmarshalIfcJSON(am, &pk, []byte(args[0])); err != nil {
```

Comment:
thinking out loud, would it be easier to changes args[0] to valConsAddress as a bech32 address?

Reply:
Yes. Not sure if someone wants to query by a public key though. Should we check about it with someone else? Who?